internal/events/event_test.go

```diff
@@ -205,390 +208,132 @@ var _ = Describe("Events library", func() {
-			infraEnv := &common.InfraEnv{
-				KubeKeyNamespace: "test_namespace",
-				InfraEnv: models.InfraEnv{
-					Name:     &infraEnvName,
-					ID:       &infraEnvId,
-					UserName: user1,
-				},
-			}
-			testHost1 := &common.Host{
-				KubeKeyNamespace: "test_namespace",
-				Host: models.Host{
-					ID:         &hostId1,
-					InfraEnvID: infraEnvId,
-					UserName:   user1,
-				},
-			}
-			testHost2 := &common.Host{
-				KubeKeyNamespace: "test_namespace",
-				Host: models.Host{
-					ID:         &hostId2,
-					InfraEnvID: infraEnvId,
-					UserName:   user1,
-				},
-			}
-			testHost3 := &common.Host{
-				KubeKeyNamespace: "test_namespace",
-				Host: models.Host{
-					ID:         &hostId3,
-					InfraEnvID: infraEnvId,
-					UserName:   user1,
-				},
-			}
-			Expect(db.Create(&c).Error).ShouldNot(HaveOccurred())
-			Expect(db.Create(&infraEnv).Error).ShouldNot(HaveOccurred())
-			Expect(db.Create(&testHost1).Error).ShouldNot(HaveOccurred())
-			Expect(db.Create(&testHost2).Error).ShouldNot(HaveOccurred())
-			Expect(db.Create(&testHost3).Error).ShouldNot(HaveOccurred())
-
-			theEvents.V2AddEvent(ctx, &clusterId, nil, &infraEnvId, "fake_event1", models.EventSeverityInfo, models.EventCategoryUser, time.Now())
-			theEvents.V2AddEvent(ctx, nil, nil, &infraEnvId, "fake_event2", models.EventSeverityInfo, models.EventCategoryUser, time.Now())
-			theEvents.V2AddEvent(ctx, nil, &hostId1, &infraEnvId, "fake_event3", models.EventSeverityInfo, models.EventCategoryUser, time.Now())
-			theEvents.V2AddEvent(ctx, nil, &hostId2, &infraEnvId, "fake_event4", models.EventSeverityInfo, models.EventCategoryUser, time.Now())
-			theEvents.V2AddEvent(ctx, nil, &hostId3, &infraEnvId, "fake_event5", models.EventSeverityInfo, models.EventCategoryUser, time.Now())
-
-			evs, err := theEvents.V2GetEvents(ctx, &clusterId, nil, &infraEnvId, models.EventCategoryUser)
-			Expect(err).Should(BeNil())
-			Expect(len(evs)).Should(Equal(1))
-			evs, err = theEvents.V2GetEvents(ctx, nil, nil, &infraEnvId, models.EventCategoryUser)
-			Expect(err).Should(BeNil())
-			Expect(len(evs)).Should(Equal(5))
-			evs, err = theEvents.V2GetEvents(ctx, nil, &hostId1, &infraEnvId, models.EventCategoryUser)
-			Expect(err).Should(BeNil())
-			Expect(len(evs)).Should(Equal(1))
-			evs, err = theEvents.V2GetEvents(ctx, &clusterId, &hostId1, nil, models.EventCategoryUser)
-			Expect(err).Should(BeNil())
-			Expect(len(evs)).Should(Equal(0))
+		Context("with user role", func() {
+			BeforeEach(func() {
+				payload := &ocm.AuthPayload{}
+				payload.Role = ocm.UserRole
+				payload.Username = "user1"
+				payload.Organization = "org1"
+				ctx = context.WithValue(context.TODO(), restapi.AuthKey, payload)
+			})
+			It("gets events on own clusters", func() {
+				By("strictly own cluster")
+				evs, err := theEvents.V2GetEvents(ctx, &cluster1, nil, nil)
+				Expect(err).ShouldNot(HaveOccurred())
+				Expect(len(evs)).To(Equal(2))
+
+				By("cluster on the same org")
```

Comment:
the same org gets 1 event where the actual owner gets 2? which event got filtered out?